src/index.js

```diff
@@ -206,10 +207,32 @@ function sanitizeHtml(html, options, _recursing) {
                 return;
               }
             }
-
+            if (name === 'iframe' && a === 'src') {
+              //Check if value contains proper hostname prefix
+              if (value.indexOf('http:') < 0 && 
```

Comment:
The presence of http: or https: somewhere else in the string, i.e. not at the beginning, isn't relevant and shouldn't cause it to be rejected. For instance a query string parameter of the string might be another URL, etc. but we're not being asked to directly fetch that.

All you need to do here is check whether it starts with `//` or not. If it starts with `//` it by definition doesn't start with one of the others.

Reply:
Agreed!